docs/data-model.md

```diff
@@ -0,0 +1,38 @@
+# Promscale Data Storage
+
+This document provides the foundations to understand how Promscale stores data.
+
+## Metrics
+
+The most important components of the metric data model are as follows:
+
+```
+                            _prom_catalog
+      ┌───────────────────────────────────────────────────────┐
+      │                                                       │
+      │   ┌──────────┐       ┌──────────┐      ┌──────────┐   │
+      │   │  metric  │◄──────┤  series  ├─────►│  label   │   │
+      │   └──────────┘       └────▲─────┘      └──────────┘   │
+      │                           │                           │
+      │                       series_id                       │
+      └───────────────────────────┼───────────────────────────┘
+                                  │
+                    ┌─────────────┴─────────────┐
+                    │  prom_data.<metric_name>  │
+                    └┬──────────────────────────┘
+                     └─ BTree (series_id, time) INCLUDE (value)
+```
+
+The `_prom_catalog` schema contains metadata tables which describe `metrics`,
+`label`s, and `series`. Each metric stored in the system is represented through
+an entry in the `metric` table. The `label` table contains (key, value) pairs
+with identity. The `series` table represents the combination of a `metric`, and
+a list of `label`s.
+
+The actual metric data is stored in a TimescaleDB hypertable (one per metric)
+with the same name as the metric. These hypertables are in the `prom_data`
```

Comment:
The name is same as metric only if less than some 64 chars limit that Postgres has. In long metric names, it can be some other name. Maybe write this as a note?

Reply:
Done.